src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpOnTypeFormattingPass.cs

```diff
@@ -84,6 +89,28 @@ public async override Task<FormattingResult> ExecuteAsync(FormattingContext cont
             return new FormattingResult(finalEdits);
         }
 
+        private TextEdit[] FilterCSharpTextEdits(FormattingContext context, TextEdit[] edits)
+        {
+            var filteredEdits = edits.Where(e => !AffectsWhitespaceInNonCSharpLine(e)).ToArray();
+            return filteredEdits;
+
+            bool AffectsWhitespaceInNonCSharpLine(TextEdit edit)
+            {
+                //
+                // Example:
+                //     @{
+                //       var x = "asdf";
+                // |  |}
+                // ^  ^ - C# formatter wants to remove this whitespace because it doesn't know about the '}'.
+                // But we can't let it happen.
+                //
+                return
+                    edit.Range.Start.Character == 0 &&
+                    edit.Range.Start.Line == edit.Range.End.Line &&
+                    !context.Indentations[(int)edit.Range.Start.Line].StartsInCSharpContext;
+            }
```

Comment:
Would this work?

```suggestion
            var filteredEdits = edits.Where(e => AffectsWhitespaceInCSharpLine(e)).ToArray();
            return filteredEdits;

            bool AffectsWhitespaceInCSharpLine(TextEdit edit)
            {
                //
                // Example:
                //     @{
                //       var x = "asdf";
                // |  |}
                // ^  ^ - C# formatter wants to remove this whitespace because it doesn't know about the '}'.
                // But we can't let it happen.
                //
                return
                    edit.Range.Start.Character == 0 &&
                    edit.Range.Start.Line == edit.Range.End.Line &&
                    context.Indentations[(int)edit.Range.Start.Line].StartsInCSharpContext;
            }
```

Reply:
It won't work without making the return statement more complicated. Right now it allows multiline edits but with your modification we will only restrict it to single line edits.